server/src/test/java/org/elasticsearch/snapshots/SnapshotsServiceTests.java

```diff
@@ -435,11 +431,11 @@ private static DiscoveryNodes discoveryNodes(String localNodeId) {
     }
 
     private static SnapshotsService.ShardSnapshotUpdate successUpdate(Snapshot snapshot, ShardId shardId, String nodeId) {
-        return new SnapshotsService.ShardSnapshotUpdate(snapshot, shardId, successfulShardStatus(nodeId));
+        return new SnapshotsService.ShardSnapshotUpdate(snapshot, shardId, null, successfulShardStatus(nodeId), null);
     }
 
     private static SnapshotsService.ShardSnapshotUpdate successUpdate(Snapshot snapshot, RepositoryShardId shardId, String nodeId) {
-        return new SnapshotsService.ShardSnapshotUpdate(snapshot, shardId, successfulShardStatus(nodeId));
+        return new SnapshotsService.ShardSnapshotUpdate(snapshot, null, shardId, successfulShardStatus(nodeId), null);
```

Comment:
For the listener could we use `ActionListener.wrap(() -> fail("should not complete publication"))` instead of `null` to make sure that any incorrect use of the listener results in an `AssertionError` that doesn't get caught and handled?